File: 프로그래머스/lv1/92334. 신고 결과 받기/신고 결과 받기.py
def solution(id_list, report, k):
    answer = [0] * len(id_list)
    
    dic = {}
    for i in id_list:
        dic[i] = 0
        
    report = set(report)    # 동일한 유저 신고 중복 제거                    
    for r in report:
        use_id, report_id = r.split(' ')    # 신고자 id, 신고 당한 id
        dic[report_id] += 1
    
    for r in report:
        use_id, report_id = r.split(' ')
        if dic[report_id] >= k:     # 신고 당한 id가 신고 횟수 이상일 때
            answer[id_list.index(use_id)] += 1  # 신고한 id가 받는 메일 수 추가
    
    return answer

File: 프로그래머스/1/92334. 신고 결과 받기/신고 결과 받기.js
function solution(id_list, report, k) {
    var answer = [];
    var report_list = {};
    
    id_list.map(id => {
        report_list[id] = [];
        answer.push(0);
    })
    
    // 신고당한 사람을 key, 신고한 사람을 value로 report_list에 넣어주기
    report.map((user) => {
        const [reporting_id, reported_id] = user.split(' ');
        if (!report_list[reported_id].includes(reporting_id)) report_list[reported_id].push(reporting_id);
    })
    
    for (let key in report_list) {
        if (report_list[key].length >= k) {     // report_list에서 신고당한 사람(key)의 신고 횟수가 k회 이상일 때
            report_list[key].map((user) => {
                // 신고한 유저에게 메일 발송(count 증가) 
                const idx = id_list.indexOf(user);
                answer[idx]++;
            })
        }
    }

    return answer;
}